src/libutil/logging.cc

```diff
@@ -184,6 +184,33 @@ struct JSONLogger : Logger {
         json["action"] = "msg";
         json["level"] = ei.level;
         json["msg"] = oss.str();
+        json["raw_msg"] = ei.hint->str();
+
+        if (ei.errPos.has_value() && (*ei.errPos)) {
+            json["line"] = ei.errPos->line;
+            json["column"] = ei.errPos->column;
+            json["file"] = ei.errPos->file;
+        } else {
+            json["line"] = -1;
+            json["column"] = -1;
+            json["file"] = nullptr;
+        }
```

Comment:
I don't think `-` is a great sentinel value. I rather do one of:

 1. Make suboject, stored at `pos`, which is nullable.
 2. Use `null` for each as the sentinel value
 3. Skip fields altogether rather than using sentinel value

in decreasing order of preference.

Reply:
@Ericson2314 I decided to set `null` by default here since AFAIU the `pos` struct *can* be undeclared, so with this we have a fair sentinel value and are on the safe side IMHO :)